### PR TITLE
add disk encryption feature based on rocksDB

### PR DESCRIPTION
**Pull request by cyjseagull**

### CODE REVIEW

File: libstorage/BasicRocksDB.cpp [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+Status BasicRocksDB::Get(ReadOptions const& options, std::string const& key, std::string& value)
+{
+    assert(m_db);
+    auto status = m_db->Get(options, Slice(std::move(key)), &value);
+    if (!status.ok() && !status.IsNotFound())
+    {
+        std::stringstream exitInfo;
+        exitInfo << "Query rocksDB failed, status:" << status.ToString();
+        ROCKSDB_LOG(ERROR) << LOG_DESC(exitInfo.str());
+        errorExit(exitInfo, QueryDBFailed());
```
Comment (wheatli) [inline libstorage/BasicRocksDB.cpp]: it will print error log twice here

File: libstorage/RocksDBStorage.cpp [hunk 2]
```diff
@@ -54,14 +54,7 @@ Entries::Ptr RocksDBStorage::select(
...
         string value;
-        auto s = m_db->Get(ReadOptions(), Slice(std::move(entryKey)), &value);
-        if (!s.ok() && !s.IsNotFound())
-        {
-            STORAGE_ROCKSDB_LOG(ERROR)
-                << LOG_DESC("Query rocksdb failed") << LOG_KV("status", s.ToString());
-
-            BOOST_THROW_EXCEPTION(StorageException(-1, "Query rocksdb exception:" + s.ToString()));
-        }
+        auto s = m_db->Get(ReadOptions(), entryKey, value);
```
Comment (wheatli) [inline libstorage/RocksDBStorage.cpp]: can not omit exception or move the exception handler into low-layer

Comment (cyjseagull) [reply libstorage/RocksDBStorage.cpp]: I have recovered this logic, but considerring the kind of status of rocksDB are more than LevelDB, this logic below is need to be checked. I will check it with xiaoBai

File: libstorage/BasicRocksDB.cpp (hunk 1 above)
Comment (cyjseagull) [reply libstorage/BasicRocksDB.cpp]: This has been replaced with checkStatus.

File: libstorage/RocksDBStorage.h [hunk 3]
```diff
@@ -20,6 +20,7 @@
  */
 #pragma once
 
+#include "BasicRocksDB.h"
```
Comment (bxq2011hust) [inline libstorage/RocksDBStorage.h]: what about use forward declare, instead include whole `"BasicRocksDB.h"`

Comment (cyjseagull) [reply libstorage/RocksDBStorage.h]: done.

**Pull request merged**